src/constraint_aware_append.c

```diff
@@ -214,7 +185,21 @@ ca_append_begin(CustomScanState *node, EState *estate, int eflags)
 				 * If this is a base rel (chunk), check if it can be
 				 * excluded from the scan. Otherwise, fall through.
 				 */
-				if (can_exclude_chunk((Scan *) plan, lfirst(lc_info), estate, restrictinfos))
+				ListCell *lc;
+				restrictinfos = NIL;
+				foreach (lc, lfirst(lc_clause))
```

Comment:
I'm confused isn't the `clause` list already a list of RestrictInfos? Why do we have to put it inside another layer of RestrictInfos here?

Reply:
no, clause is the expression of the restrictinfo, restrictinfo is planner only and cant be serialized so it has to be handled this way